Source/ACE/Managers/RecipeManager.cs

```diff
@@ -0,0 +1,135 @@
+                        {
+                            if (source.Structure <= 1)
+                                player.DestroyInventoryItem(source);
+                            else
+                                source.Structure--;
+                        }
+
+                        if ((recipe.ResultFlags & (uint)RecipeResult.TargetItemUsesDecrement) > 0)
+                        {
+                            if (target.Structure <= 1)
+                                player.DestroyInventoryItem(target);
+                            else
+                                target.Structure--;
+                        }
+
+                        if (success)
+                        {
+                            WorldObject newObject1 = null;
+                            WorldObject newObject2 = null;
+
+                            if ((recipe.ResultFlags & (uint)RecipeResult.CreateNewItem1) > 0 && recipe.Item1Wcid != null)
+                            {
+                                newObject1 = Factories.WorldObjectFactory.CreateNewWorldObject(recipe.Item1Wcid.Value);
+                                newObject1.ContainerId = player.Guid.Full;
+                                newObject1.Placement = 0;
+                                player.HandleAddToInventoryEx(newObject1);
+                            }
+
+                            if ((recipe.ResultFlags & (uint)RecipeResult.CreateNewItem2) > 0 && recipe.Item2Wcid != null)
+                            {
+                                newObject2 = Factories.WorldObjectFactory.CreateNewWorldObject(recipe.Item2Wcid.Value);
+                                newObject2.ContainerId = player.Guid.Full;
+                                newObject2.Placement = 0;
+                                player.HandleAddToInventoryEx(newObject2);
+                            }
+
+                            var text = string.Format(recipe.SuccessMessage, source.Name, target.Name, newObject1.Name, newObject2?.Name);
+                            var message = new GameMessageSystemChat(text, ChatMessageType.Craft);
+                            player.Session.Network.EnqueueSend(message);
+                        }
+
+                        break;
+                    case RecipeType.Dyeing:
+                        break;
+                    case RecipeType.Healing:
+                        break;
+                    case RecipeType.ManaStone:
+                        break;
+                    case RecipeType.Tinkering:
+                        break;
+                    case RecipeType.Unlocking:
+                        break;
+                    case RecipeType.None:
+                        return;
+                }
+            });
+
+            craftChain.AddAction(player, () => player.HandleActionMotion(new UniversalMotion(MotionStance.Standing, new MotionItem(MotionCommand.Ready))));
```

Comment:
These are performance, not correctness -- fix if you wish...

You may consider coalescing this action with the prior one in the chain.

Your also calling player.HandleAction<thing> from within the Player's action.  HandleAction functions by convention create their own actions, so your doubling down on action creation here (this will effectively spend the player's next action phase creating the action, then running the action in the phase after that.  Ideally you would run the action in the next phase).

Reply:
Yes, its fine to call then (although I would still coalesce the call into the prior action on the chain).  I screwed up naming it -- It should have been something like "HandleMotion".  I should fix that in a future PR :-/.